zimbra/src/main/resources/public/ts/controllers/controller.ts

```diff
@@ -980,18 +983,15 @@ export let zimbraController = ng.controller("ZimbraController", [
             );
         };
 
-        $scope.postAttachments = async () => {
-            const mail = $scope.state.newItem as Mail;
-            if (!mail.id) {
-                await Zimbra.instance.folders.draft.saveDraft(mail);
-                await mail.postAttachments($scope);
-            } else {
-                await mail.postAttachments($scope);
-            }
-        };
-
         $scope.deleteAttachment = function(event, attachment, mail) {
-            mail.deleteAttachment(attachment);
+            const response = mail.deleteAttachment(attachment);
```

Comment:
il manquerait pas un await ?

Reply:
quand j'essaie de le mettren intelliJ me dit que ce n'est pas bon : 

TS1308: 'await' expression is only allowed within an async function.

Reply:
bah tu rends la fonction async ce n'est pas grave

Reply:
le problème c'est que la variable response ne va pas attendre le résultat de deleteAttachment avant de passer à la suite